script.js

```diff
@@ -1,12 +1,377 @@
+  pageNavigation.prepend(selectAnEpisode);
+
+  let getEpisodesTitles = document.getElementsByClassName("episodeName")
+
+  let episodeOption = document.createElement("option");
+  episodeOption.innerText = "All Results";
+  selectAnEpisode.appendChild(episodeOption);
+
+  for (let titleIndex = 0; titleIndex < getEpisodesTitles.length; titleIndex++) {
+    episodeOption = document.createElement("option");
+    selectAnEpisode.appendChild(episodeOption);
+
+    episodeOption.innerText = getEpisodesTitles[titleIndex].innerText;
+  }
+
+  selectAnEpisode.addEventListener("change", selectedEpisode)
+}
+
+function selectedEpisode(e) {
+  let selectedOption = e.target.value;
+  let getResultsOnPage = document.getElementsByClassName("wrapper");                             //I declared this variable again (there is one in level 200), did not want to have anything in the global scope, so not sure if it is best practice?
+  let allResultsOnPage = Array.from(getResultsOnPage);
+
+  allResultsOnPage.forEach(result => {
+    if (result.innerText.includes(selectedOption)) {
+      result.classList.remove("hidden");
+    } else {
+      result.classList.add("hidden");
+    }
+
+    if (selectedOption === "All Results") {
+      result.classList.remove("hidden");
+    }
+
+  })
+}
+
+//Level 400
+
+function addShowsSelector() {
+  let selectAShow = document.createElement("select");
+  selectAShow.setAttribute("id", "shows");
+
+  let chooseEpisodeAndShow = document.createElement("label");
+  chooseEpisodeAndShow.innerText = "Choose a show";
+  chooseEpisodeAndShow.classList.add("chooseLabel")
+
+  let pageNavigation = document.querySelector("nav")
+  pageNavigation.prepend(chooseEpisodeAndShow, selectAShow);
+
+  let showsList = getAllShows();
+
+  let sortedShowsList = showsList.sort(function compare(showA, showB) {
+    if (showA.name.toUpperCase() < showB.name.toLocaleUpperCase()) { return -1 }
+    if (showA.name.toUpperCase() > showB.name.toUpperCase()) { return 1 }
+    return 0;
+  })
+
+  for (let showIndex = 0; showIndex < sortedShowsList.length; showIndex++) {
+    let showOption = document.createElement("option");
+    selectAShow.appendChild(showOption);
+
+    showOption.innerText = showsList[showIndex].name
+    showOption.setAttribute("value", `${showsList[showIndex].id}`)
+  }
+
+  selectAShow.addEventListener("change", selectedShow)
+}
+
+function selectedShow(e) {
+  showID = +e.target.value
+
+  removeCurrentPage()
+  getAllEpisodes(showID)
+}
+
+function removeCurrentPage() {
+  let rootElem = document.getElementById("root");
+  rootElem.innerHTML = "";
+
+  let navigationWrapper = document.querySelector(".navigationWrapper")
+  navigationWrapper.innerHTML = "";
+}
+
+
+//Level 500 part 1 and 2
+
+function sortedShowsList() {
+  let showsList = getAllShows();
+
+  let newShowsList = showsList.sort(function compare(showA, showB) {
```

Comment:
I'm gonna pick at the naming of the variable here. 

I think it would be better to name your variable in a way that sets the expectation of what you're storing in the variable. In this case, I'd expect that `newShowsList` will store a sorted list. So, I think naming the variable `sortedShowList` would be better.

I'm guessing that you didn't name it that because you already named your function `sortedShowsList`. For functions, I think you should use a verb to start the name. The reason is that most functions will complete an action, so you should use a verb to indicate that in the name. So, a better name for the function would be `sortShowList` (the verb here is `sort`).

You can read more here: https://dmitripavlutin.com/coding-like-shakespeare-practical-function-naming-conventions/#2-prefer-explanatory-names